File: scratch/generate_splashes.js
const fs = require('fs');
const path = require('path');
const Jimp = require('jimp');

const logoPath = path.join(__dirname, '..', 'public', 'LaHustle-Original.png');

const splashSizes = [
    { w: 1290, h: 2796 },
    { w: 1179, h: 2556 },
    { w: 1125, h: 2436 },
    { w: 1242, h: 2688 },
    { w: 1668, h: 2388 },
    { w: 2048, h: 2732 },
    { w: 2732, h: 2732 },
    { w: 640,  h: 1136 },
    { w: 750,  h: 1334 },
    { w: 828,  h: 1792 }
];

async function generate() {
    console.log('Loading logo from:', logoPath);
    const logo = await Jimp.read(logoPath);
    
    for (const size of splashSizes) {
        console.log(`Generating splash ${size.w}x${size.h}...`);
        
        // Create background (#050505 in hex color format for Jimp: 0x050505FF)
        const background = new Jimp(size.w, size.h, 0x050505FF);
        
        // Clone logo and resize
        const tempLogo = logo.clone();
        
        // Target logo width is 60% of screen width
        const targetWidth = Math.round(size.w * 0.60);
        tempLogo.resize(targetWidth, Jimp.AUTO);
        
        // Center position
        const x = Math.round((size.w - tempLogo.bitmap.width) / 2);
        const y = Math.round((size.h - tempLogo.bitmap.height) / 2);
        
        // Composite
        background.composite(tempLogo, x, y);
        
        const outputPath = path.join(__dirname, '..', 'public', `splash-${size.w}x${size.h}.png`);
        await background.writeAsync(outputPath);
        console.log(`Saved: ${outputPath}`);
    }
    console.log('All splash screens generated successfully!');
}

generate().catch(console.error);

File: scratch/generate_splashes.py
import os
from PIL import Image

logo_path = os.path.join("public", "LaHustle-Original.png")

splash_sizes = [
    (1290, 2796),
    (1179, 2556),
    (1125, 2436),
    (1242, 2688),
    (828, 1792),
    (750, 1334),
    (2048, 2732),
    (1668, 2388),
    (2732, 2732),
    (640, 1136)
]

def generate():
    print(f"Loading logo from: {logo_path}")
    logo = Image.open(logo_path)
    
    for width, height in splash_sizes:
        print(f"Generating splash {width}x{height}...")
        
        # Create solid background #050505
        background = Image.new("RGBA", (width, height), (5, 5, 5, 255))
        
        # Calculate new logo size (60% of screen width)
        target_width = int(width * 0.60)
        aspect_ratio = logo.height / logo.width
        target_height = int(target_width * aspect_ratio)
        
        # Resize logo using high-quality resampling
        resized_logo = logo.resize((target_width, target_height), Image.Resampling.LANCZOS)
        
        # Center logo
        x = (width - target_width) // 2
        y = (height - target_height) // 2
        
        # Paste logo using alpha channel as mask if logo has alpha
        if resized_logo.mode in ('RGBA', 'LA') or (resized_logo.mode == 'P' and 'transparency' in resized_logo.info):
            background.paste(resized_logo, (x, y), resized_logo)
        else:
            background.paste(resized_logo, (x, y))
            
        output_path = os.path.join("public", f"splash-{width}x{height}.png")
        background.convert("RGB").save(output_path, "PNG")
        print(f"Saved: {output_path}")
        
    print("All splash screens generated successfully!")

if __name__ == "__main__":
    generate()
